### ISSUE TITLE
Migrate to Java 11

### ISSUE THREAD

**Issue by lucasvalenteds** [lucasvalenteds/responder#2]
Java 10 brings benefits to Container-based applications (Docker, for instance), mainly better control over JVM resources usage like RAM allocation and CPUs.

**Known issues**
* None alpine-based image available on Docker Hub ([issue](https://github.com/docker-library/openjdk/issues/100#issuecomment-282149214))
* MockK stopped working on `RouterTest` class
```text
  io.responder.server.RouterTest

  Test It binds all endpoints to a RouterFunction FAILED (3.6s)

  io.mockk.MockKException: Can't instantiate proxy for class kotlin.Any
      at io.responder.server.RouterTest.testBindRoutes(RouterTest.kt:25) 
```

* Gatling does not work with Java 11 yet (Version 3 will support it)
```text
> Exception in thread "main" io.gatling.core.session.el.ElParserException: Failed to parse application/json with error 'class [B cannot be cast to class [C ([B and [C are in module java.base of loader 'bootstrap')'
```

**References**
* Blog post [Java & Docker: Java 10 improvements strengthen the friendship!](https://aboullaite.me/docker-java-10/)

**Comment by adriaanm** [comment]
On the Scala side, although not officially supported, recent versions of the compiler (2.12.6, but also the last minor releases of 2.10 and 2.11) should run fine on 9, 10 and 11. If not, please let us know at the linked ticket!

**Comment by lucasvalenteds** [comment]
Hi @adriaanm. Thank you so much for commenting!

The code compiles, as you've mentioned, but there is an exception thrown at runtime which should be fixed in Gatling 3. Sorry about the confusion, the issue description was updated to clarify the known issues.

**Comment by lucasvalenteds** [comment]
closed b6581598030fa99564050fcf73fcded13874c245
